assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * `modules/analytics-4` data store: properties.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const fetchGetPropertiesStore = createFetchStore( {
+	baseName: 'getProperties',
+	controlCallback( { accountID } ) {
+		return API.get( 'modules', 'analytics-4', 'properties', { accountID }, {
+			useCache: true,
+		} );
+	},
+	reducerCallback( state, properties, { accountID } ) {
+		return {
+			...state,
+			properties: {
+				...state.properties,
+				[ accountID ]: properties,
+			},
+		};
+	},
+	argsToParams( accountID ) {
+		return { accountID };
+	},
+	validateParams( { accountID } = {} ) {
+		invariant( accountID, 'GA4 accountID is required.' );
+	},
+} );
+
+const fetchCreatePropertyStore = createFetchStore( {
+	baseName: 'createProperty',
+	controlCallback( { accountID } ) {
+		return API.set( 'modules', 'analytics-4', 'create-property', { accountID } );
+	},
+	reducerCallback( state, property, { accountID } ) {
+		return {
+			...state,
+			properties: {
+				...state.properties,
+				[ accountID ]: [
+					...( state.properties[ accountID ] || [] ),
+					property,
+				],
+			},
+		};
+	},
+	argsToParams( accountID ) {
+		return { accountID };
+	},
+	validateParams( { accountID } = {} ) {
+		invariant( accountID, 'GA4 accountID is required.' );
+	},
+} );
+
+const baseInitialState = {
+	properties: {},
+};
+
+const baseActions = {
+	/**
+	 * Creates a new GA4 property.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} accountID GA4 account ID.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*createProperty( accountID ) {
+		invariant( accountID, 'GA4 accountID is required.' );
```

Comment:
`invariant` won't work as expected within a generator function. See https://github.com/google/site-kit-wp/issues/1941

Here, we'd have to do something like this: https://github.com/google/site-kit-wp/blob/7a0002b13d3a1d4b2df69da424183faaa843c4e5/assets/js/googlesitekit/datastore/user/feature-tours.js#L99-L102